airavata-api/airavata-api-server/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java

```diff
@@ -692,6 +695,80 @@ public String generateAndRegisterSSHKeys(AuthzToken authzToken, String descripti
+                AiravataSystemException exception = new AiravataSystemException(AiravataErrorType.INTERNAL_ERROR);
+                exception.setMessage("Experiment not configured with a Group Resource Profile: " + experiment.getExperimentId());
+                throw exception;
+            }
+
+            GatewayResourceProfile gatewayResourceProfile = regClient.getGatewayResourceProfile(gatewayId);
+            String token;
+            if(isValid(storagePreference.getResourceSpecificCredentialStoreToken()))
+                token = storagePreference.getResourceSpecificCredentialStoreToken();
+            else
+                token = gatewayResourceProfile.getCredentialStoreToken();
+
+            StorageResourceAdaptor adaptor = AdaptorSupportImpl.getInstance().fetchStorageAdaptor(
+                    gatewayId,
+                    storageResourceId,
+                    DataMovementProtocol.SCP,
+                    token,
+                    storagePreference.getLoginUserName());
+            String experimentDataDir = experiment.getUserConfigurationData().getExperimentDataDir();
+            String userDirectory = getUserDirectory(experimentDataDir);
+
+            //shouldn't retrieve size in giga bytes since the du -s command rounds off the value to a higher bound.
+            CommandOutput output = adaptor.executeCommand("du -sm", userDirectory);
+            if(!output.getStdError().isEmpty()) {
+                logger.error("The experiment data directory + " + experimentDataDir + " configured on the experiment is wrong.");
+                AiravataSystemException exception = new AiravataSystemException(AiravataErrorType.INTERNAL_ERROR);
+                exception.setMessage("The experiment data directory + " + experimentDataDir + " configured on the experiment is wrong.");
+                throw exception;
+            }
+
+            double userDirectorySize = Double.parseDouble(output.getStdOut().substring(0, output.getStdOut().length() - 2).trim())/1024.0;
+            if(userDirectorySize >= storagePreference.getUserStorageQuota()) {
+                logger.error(String.format("The user directory %s has exceeded the storage quota: %d GB", userDirectory, storagePreference.getUserStorageQuota()));
+                AiravataSystemException exception = new AiravataSystemException();
+                exception.setMessage(String.format("The user directory %s has exceeded the storage quota: %d GB", userDirectory, storagePreference.getUserStorageQuota()));
+                throw exception;
+            }
+        } catch (Exception ex) {
+            logger.error(ex.getMessage(), ex);
+            AiravataSystemException exception = new AiravataSystemException(AiravataErrorType.INTERNAL_ERROR);
+            exception.setMessage(ex.getMessage());
+            throw exception;
+        }
+    }
+
+    private String getUserDirectory(String experimentDataDirectory) {
```

Comment:
This is a blind algorithm. At least provide your assumptions for experimentDataDirectory value as comments